raptor/pipeline.md

```diff
@@ -0,0 +1,103 @@
+    - Tool to verify whether generated code snippets compile
+    - Creates compilation test cases for each V1 and Beta snippet live at docs.microsoft.com
+    - Consumes Microsoft.Graph V1 and Beta DLLs as Nuget packages
+3. [microsoftgraph/msgraph-sdk-dotnet](https://github.com/microsoftgraph/msgraph-sdk-dotnet)
+    - SDK repo to compile generated code
+4. [microsoftgraph/microsoft-graph-explorer-api](https://github.com/microsoftgraph/microsoft-graph-explorer-api)
+    - Provides an HTTP endpoint to generate C# snippets from HTTP snippets. The HTTP endpoint is the only current option that is exposed for this functionality.
+5. [microsoftgraph/microsoft-graph-docs](https://github.com/microsoftgraph/microsoft-graph-docs)
+    - Contains HTTP snippets
+    - New documentation can be added by forking a branch
+6. [OneDrive/apidoctor](https://github.com/OneDrive/apidoctor)
+    - Tool to extract HTTP snippets from docs and write generated snippet references
+    - Snippet generation functionality is not available in the main repo.
+      - It is available at [andrueastman/apidoctor](https://github.com/andrueastman/apidoctor) / [generate-snippets](https://github.com/andrueastman/apidoctor/tree/generate-snippets).
+      - The functionality is coupled with a PR generation to docs repo upon completion of snippet generation.
+      - Hits public Graph Explorer API HTTP endpoint to generate the snippets
+
+## Proposed Solution
+### Input variables
+1. `version` : `v1.0` or `beta`
+2. either `metadataURL` or metadata artifact from another pipeline run
+3. `docsBranch` : branch on `microsoft-graph-docs` repo, containing new documentation
+4. `typewriterReleaseVersion` : true if the released version of typewriter to be used from Github releases. Otherwise it is expected to be compiled from source.
+
+### Pipeline
+  1. Prepare metadata input
+     1. If metadata is given as a public URL in `metadataURL`, set `metadataPath` variable to that
+        1. Typewriter can take URL as metadata input
+     1. If metadata is an Azure DevOps artifact
+        1. Download the artifact to a local path
+        2. Save that local path as `metadataPath`
+  2. Build snippet generation tool
+     1. Checkout `microsoft-graph-explorer-api` repo
+     2. Build snippet generation tool
+     3. Save executable's path as `snippetGeneratorPath`
+  3. Checkout `microsoft-graph-docs` repo
+     1. Use `docsBranch` variable if defined
+     2. Use `master` if not defined
+     3. Save repo path as `docsRepoPath`
+  4. Run apidoctor
+     1. checkout `OneDrive/apidoctor` repo
+     2. run `apidoctor` with the following inputs
+        1. `docsRepoPath`
+        2. `snippetGeneratorPath`
+  5. SDK generation
+     1. checkout `msgraph-sdk-dotnet` repo
+        1. Save path as `dotnetRepoPath`
+     2. If `typewriterReleaseVersion` is `true`
+        1. Download `Typewriter` from Github releases
+     3. Else
+        1. Checkout `MSGraph-SDK-Code-Generator` repo
```

Comment:
I assume this would be the dev branch, correct?

Reply:
Yeah, since the main branch is expected to be in sync with the Github release, it makes sense to use `dev` as the configuration here. For manual runs, it can be picked at runtime using the Azure DevOps interface.